src/main/java/org/opensearch/flowframework/workflow/DeleteAgentStep.java

```diff
@@ -85,9 +87,21 @@ public void onResponse(DeleteResponse deleteResponse) {
                 @Override
                 public void onFailure(Exception ex) {
                     Exception e = getSafeException(ex);
-                    String errorMessage = (e == null ? "Failed to delete agent " + agentId : e.getMessage());
-                    logger.error(errorMessage, e);
-                    deleteAgentFuture.onFailure(new WorkflowStepException(errorMessage, ExceptionsHelper.status(e)));
+                    // NOT_FOUND exception should be treated as successful deletion
```

Comment:
it would be good to double check what ml-commons returns in case when user has no permission to access the model when deleting. Either manually or through code

Reply:
@amitgalitz 
[Delete Model](https://github.com/opensearch-project/ml-commons/blob/5ad26df7dd23a6ed37a13d80548779579521b13f/plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java) returns NOT_FOUND if a get request for that model has false `isExists()`.  It returns FORBIDDEN for no permissions.

[Delete Agent](https://github.com/opensearch-project/ml-commons/blob/5ad26df7dd23a6ed37a13d80548779579521b13f/plugin/src/main/java/org/opensearch/ml/action/agents/DeleteAgentTransportAction.java) does the same.